### PR TITLE
ci: add job to keep master green

### PR DESCRIPTION
**Pull request by qzhuyan**
example run:
https://github.com/emqx/emqx/actions/runs/4477786641/jobs/7869749635

Just to be clear, this is not trying to hide the issue, next PR would be to find all the flaky TCs from history reruns on master branch.

## Checklist for CI (.github/workflows) changes

~If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)~
~Change log has been added to `changes/` dir for user-facing artifacts update~

### CODE REVIEW

**Review by ieQu1 — APPROVED** [review]

File: .github/workflows/geen_master.yaml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: run script
+        shell: bash
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          python3 scripts/rerun-failed-checks.py
```
Comment (id) [inline .github/workflows/geen_master.yaml:27]: you can give $GITHUB_SHA to the script as argument here and skip fetching latest commit from github api

Comment (qzhuyan) [reply .github/workflows/geen_master.yaml:27]: I want to be flexible on the branch name to trigger.

File: .github/workflows/geen_master.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  rerun-failed-jobs:
+    runs-on: ubuntu-22.04
+    if: github.repository_owner == 'emqx'
+    permissions:
+      checks: read
+      actions: write
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```
[the remark this answers is not in the document]
Comment (qzhuyan) [reply .github/workflows/geen_master.yaml:21]: the script is in the source tree. or you mean the `fetch-depth: 0`?

[the remark this answers is not in the document]
Comment (qzhuyan) [reply .github/workflows/geen_master.yaml:21]: Oh I didn't know `fetch-depth: 0` means all history...., thought it is on the opposite.

**Pull request merged**